Feature/multiple locations api support

rflinnenbank opened the pull request:
Added Weather Context Query version for Multiple locations.

Note that this only enables the option, whereas the retrieved file formats may or may not be optimal for a large number of locations.

**For Example:**
A full request of KNMI "Harmonie Arome" weather factors for 9 predictions and 113 locations was tested as functional, but it did take about half an hour to process the full (originally 3.6Gb data) and select the needed information, and the resulting NetCDF4 file was about 77 Mb.

rflinnenbank committed:
[2022-12-05 14:32] Multi-location requests are working

TODO:
- Increase response time by changing the passthrough method to better deal with multiple locations
- Fix Factor selection (currently no factor selection takes place...)

Signed-off-by: Raoul Linnenbank <58594297+rflinnenbank@users.noreply.github.com>

rflinnenbank committed:
[2022-12-05 14:32] Multi-location requests are working

TODO:
- Increase response time by changing the passthrough method to better deal with multiple locations
- Fix Factor selection (currently no factor selection takes place...)

Signed-off-by: Raoul Linnenbank <58594297+rflinnenbank@users.noreply.github.com>

johandebraak reviewed (approved):
Ziet er prima uit. M.i. één inline comment even aanpassen en dan is het goed.

rflinnenbank committed:
[2022-12-12 10:18] Multi-location requests are working

CHANGES:
 - Changed the comment for the first day of the repository to correspond to the actual moment in the value.
Signed-off-by: Raoul Linnenbank <58594297+rflinnenbank@users.noreply.github.com>

rflinnenbank commented on weather_provider_api/routers/weather/sources/knmi/client/arome_repository.py:
```diff
@@ -66,7 +66,7 @@ def _get_repo_sub_folder(self):
     @property
     def first_day_of_repo(self):
-        first_day_of_repo = datetime.utcnow() - relativedelta(years=1)  # A year back
+        first_day_of_repo = datetime.utcnow() - relativedelta(years=3)  # A year back
```
Adjusted the inline comment accordingly!

johandebraak reviewed (commented):
top!
